addons/product/static/src/product_catalog/search/search_panel.js

```diff
@@ -1,54 +1,187 @@
+        } else if (hotkey === 'escape') {
+            this.state.isAddingSection = false;
+            this.state.newSectionName = "";
+        }
+    }
+
+    setSelectedSection(sectionId=null, filtered=false) {
+        this.env.searchModel.setSelectedSection(sectionId, filtered);
+    }
+
+    async createSection() {
+        const sectionName = this.state.newSectionName.trim();
+        if (!sectionName) return this.state.isAddingSection = false;
+
+        const section = await rpc('/product/catalog/create_section',
+            this._getSectionInfoParams({section_name: sectionName})
+        );
+
+        if (section) {
+            this.state.sectionOfSections.set(section.id, {
+                name: section.name,
+                sequence: section.sequence,
+                line_count: 0,
+            });
+            this.setSelectedSection(section.id);
+        }
+        this.state.isAddingSection = false;
+        this.state.newSectionName = "";
+    }
+
+    async loadSections() {
+        if (!this.isNonMrpOrderActive) return;
+        const sections = await rpc('/product/catalog/get_sections', this._getSectionInfoParams());
+
+        const sectionMap = this.state.sectionOfSections || new Map();
+        for (const {id, name, sequence, line_count} of sections) {
+            const existingSection = sectionMap.get(id);
+            if (!existingSection) {
+                sectionMap.set(id, {name, sequence, line_count});
+            }
+        }
+        this.state.sectionOfSections = sectionMap;
+    }
+
+    async reorderSections(moveId, targetId) {
+        [moveId, targetId] = [parseInt(moveId), parseInt(targetId)];
+        const sections = this.state.sectionOfSections;
+        const moveSection = sections.get(moveId);
+        const targetSection = sections.get(targetId);
+
+        if (!moveSection || !targetSection) return;
+
+        const updatedSequences = await rpc('/product/catalog/resequence_sections',
+            this._getSectionInfoParams({
+                sections: [
+                    { id: moveId, sequence: moveSection.sequence },
+                    { id: targetId, sequence: targetSection.sequence },
+                ],
+            })
+        );
+        for (const [id, sequence] of Object.entries(updatedSequences)) {
+            const section = sections.get(parseInt(id));
+            if (section) {
+                section.sequence = sequence;
+            }
+        }
+        this.state.sectionOfSections = new Map(
+            [...sections].sort((a, b) => a[1].sequence - b[1].sequence)
+        );
+    }
+
+    _getSectionInfoParams(extra = {}) {
+        const ctx = this.env.model.config.context;
+        return {
+            res_model: ctx.product_catalog_order_model,
+            order_id: ctx.order_id,
+            child_field: ctx.child_field,
+            ...extra,
+        };
+    }
+
+    updateSectionLineCount(event) {
+        const {sectionId, lineCountChange} = event.detail;
```

Comment:
```suggestion
    updateSectionLineCount({ detail: { sectionId, lineCountChange } }) {
```
Haven't tested this one, but I guess this should work.